test/functional/ignore_undefined.test.js

```diff
@@ -148,4 +149,128 @@ describe('Ignore Undefined', function () {
       });
     }
   });
+
+  it('Should correctly inherit ignore undefined field from db during insert', {
+    metadata: { requires: { topology: ['single'] } },
+
+    test: function (done) {
+      var configuration = this.configuration;
+      var client = configuration.newClient(configuration.writeConcernMax(), {
+        poolSize: 1,
+        ignoreUndefined: false
+      });
+
+      client.connect(function (err, client) {
+        var db = client.db(configuration.db, { ignoreUndefined: true });
+        var collection = db.collection('shouldCorrectlyIgnoreUndefinedValue3');
+
+        // Ignore the undefined field
+        collection.insert({ a: 1, b: undefined }, configuration.writeConcernMax(), err => {
+          expect(err).to.not.exist;
+
+          // Locate the doument
+          collection.findOne((err, item) => {
+            test.equal(1, item.a);
+            test.ok(item.b === undefined);
+            client.close(done);
+          });
+        });
+      });
+    }
+  });
+
+  it(
+    'Should correctly inherit ignore undefined field from collection during insert',
+    withClient(function (client, done) {
```

Comment:
Nope, there is no reason! I guess I didn't see exactly why that was needed, so I didn't include it. Will make this change in case there is something I am missing.

Reply:
I think in some cases we limit very basic tests that don't depend on replica sets or sharded clusters to just run on the `single` topology, because we have extra tests that we only run on replica/sharded, so those tasks take longer to complete in Evergreen. Limiting tests to single when possible is a way we can cut down on the wait time (and cost!) of running our tests. @mbroadst might be able to provide more background about this.

Reply:
Oops, I just left a comment about this below. I think the rule of thumb should actually be that we run tests everywhere when possible, so we maximize our coverage. Let's tackle the runtime of the tests at a later date, we can likely make a number of improvements to speed things up while maintaining as much coverage as possible across multiple cluster types